platforms/common/include/px4_platform_common/px4_work_queue/WorkItem.hpp

```diff
@@ -57,6 +60,9 @@ class WorkItem : public ListNode<WorkItem *>, public IntrusiveQueueNode<WorkItem
 	WorkItem(WorkItem &&) = delete;
 	WorkItem &operator=(WorkItem &&) = delete;
 
+	// WorkItems sorted by name
+	bool operator<=(const WorkItem &rhs) const { return (strcmp(ItemName(), rhs.ItemName()) <= 0); }
```

Comment:
Should this have an upper bound for safety, so with `strncmp`?

Reply:
I suppose, although in a lot of these cases for us I think we'd want to go back to the source to get a reasonable answer in place for the max length. 

In this case the `WorkItem` name is set in the constructor and I believe we're currently always setting it to `MODULE_NAME` or `<single_shot>`. `MODULE_NAME` doesn't really have any constraints, although in the past we've trimmed them to fit within the NuttX task `CONFIG_TASK_NAME_SIZE`. 

I'm not sure if we need to worry about about it unless there's the possibility of involving a non-NULL terminated character array.